ui/guest/src/react/ZoneMarker.tsx

```diff
@@ -117,11 +133,27 @@ export function ZoneMarker(props: ZoneMarkerProps) {
         >
           <Paper className={classes?.paper} sx={sx.paper}>
             <div style={{ display: 'flex' }}>
-              {inherited ? <LevelDescriptorIcon sx={sx.icon} /> : <FieldIcon sx={sx.icon} />}
+              {isLockedItem || isStale ? (
+                <LockedStateIcon />
+              ) : inherited ? (
+                <LevelDescriptorIcon sx={sx.icon} />
+              ) : (
+                <FieldIcon sx={sx.icon} />
+              )}
               <Typography title={label} noWrap sx={{ pointerEvents: 'all' }}>
                 {label}
               </Typography>
             </div>
+            {isLockedItem && (
+              <Typography noWrap variant="body2" component="div">
+                <FormattedMessage defaultMessage="Locked by {username}" values={{ username: lockInfo.username }} />
+              </Typography>
+            )}
+            {isStale && (
+              <Typography noWrap variant="body2" component="div">
+                <FormattedMessage defaultMessage="Item was modified. Refresh to enable editing." />
+              </Typography>
+            )}
```

Comment:
FormatJS is not setup on XB. Does this even work? Doesn't produce runtime exceptions?

Reply:
I didn't see exceptions, reverted